File: 26 - Months to Pay Off a Credit Card/Golang/main.go
// Package main provides ...
package main

import (
	"fmt"
	"math"
)

func months_to_pay_off(i, b, p float64) float64 {
	nominator := math.Log10(1+b/p*(1-math.Pow(1+i, 30))) * (-1.0 / 30)
	return nominator / math.Log10(1+i)
}

func main() {
	var balance, apr, montly float64
	fmt.Print("What is your balance? ")
	fmt.Scanf("%f", &balance)
	fmt.Print("What is the APR on the card (as a percent)? ")
	fmt.Scanf("%f", &apr)
	fmt.Print("What is the montly payment you can make? ")
	fmt.Scanf("%f", &montly)

	months := math.Ceil(months_to_pay_off(apr/100/365, balance, montly))

	fmt.Printf("It will take you %0.f months to pay off this card.", months)
}

File: 26 - Months to Pay Off a Credit Card/C++/main.cpp
#include <cmath>
#include <iostream>

double months_to_pay_off(double i, double b, double p) {
    const double nominator =
        log10(1.0 + b / p * (1.0 - pow(1.0 + i, 30))) * -1.0 / 30;
    return nominator / log10(1 + i);
}

int main() {
    double balance, apr, monthly = 0;
    std::cout << "What is your balance? ";
    std::cin >> balance;
    std::cout << "What is the APR on the card (as a percent)? ";
    std::cin >> apr;
    std::cout << "What is the monthly payment you can make? ";
    std::cin >> monthly;

    double months = ceil(months_to_pay_off(apr / 100 / 365, balance, monthly));

    std::cout << "It will take you " << months
              << " months to pay off this card." << std::endl;
    return 0;
}
